Order slips - ignore order discounts in taxes breakdown

shaffe-fr opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | As order slips do not use the discounts of the order (http://forge.prestashop.com/browse/PSCSX-8967), the product taxes breakdown shouldn't use them neither. By reseting the order discount, we make sure that the product taxes breakdown only use the totals computed in the HTMLTemplateOrderSlip::getContent method.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #18742
| How to test?  | 1. Create an order with 2 products (total of 100 € with a 20% VAT) and a 10 € discount<br>2. Create an order slip for 1 product (45 €)<br>3. Open the order slip PDF<br>4.The taxes breakdown total taxes should be 9 € (`45 * 0.2`) but it is 7 € (`(45 - 10) * 0.2`).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/17719)
<!-- Reviewable:end -->

shaffe-fr commented:
I created an issue that explains the problem.

Progi1984 commented on classes/pdf/HTMLTemplateOrderSlip.php:
```diff
@@ -274,10 +274,13 @@ protected function getTaxBreakdown()
         $breakdown = [];
 
+        // Ignore order's discounts
+        $this->order->total_discounts_tax_excl = $this->order->total_discounts_tax_incl = 0;
```
@shaffe-fr May be store in a temporary variable value for these variables and reassign it after `$this->order->getProductTaxesDetails`. No ?

shaffe-fr replied:
Yes, thanks! It's fixed.

shaffe-fr committed:
fix(order slips): ignore discounts in taxes breakdown

As order slips do not use the discounts of the order (http://forge.prestashop.com/browse/PSCSX-8967), the product taxes breakdown shouldn't use them neither.

marionf commented:
Hello @shaffe-fr
As indicated in [the issue](https://github.com/PrestaShop/PrestaShop/issues/18742#issuecomment-621952699), we couldn't reproduce the issue with the 1.7.7, so we guess it has already been fixed during the migration of the order page in symfony.
For these reasons I close your PR

Prestaplugins commented:
Hello, I still have the problem on 1.7.7.3
I have an order with a voucher for free shipping.
I make a partial refund with an order slip for a product.
The tax displayed in the order slip is totally wrong because the tax applied to the freeshipping voucher is used during the calculation...

Prestaplugins commented:
![image](https://user-images.githubusercontent.com/3033446/116281070-9cebf680-a789-11eb-95ad-1962b40377eb.png)
